Classes/Outlet/AbstractOutlet.php

```diff
@@ -132,10 +157,15 @@ public function addPipeOut(PipeInterface $pipe)
      */
     public function fill($data)
     {
+        $this->validate($data);
         foreach ($this->pipesIn as $pipe) {
+            if ($pipe instanceof ViewAwarePipeInterface) {
```

Comment:
well, we could have pipes here, that do not need a view and yet might be useful, so i think this behavior should be the way to go